src/lib/request.js

```diff
@@ -88,12 +88,19 @@ export default function (relativeUrl, opts, callback) {
 
     return fetch(url, config)
         .then((response) => {
-            if (response.headers.get('content-type').indexOf('application/json') !== -1) {
-                return response.json();
+            let result = null;
+
+            if (response.status >= 400 && response.status < 500) {
```

Comment:
So 401 is a pretty general code, not sure if we actually want to trigger a reload in all of these cases. So I'm open to suggestions on how to make this more specific for the preview logic, I can include a specific error message or response header we can match against from bcapp, for example.